src/pytorch_lightning/core/saving.py

```diff
@@ -202,15 +202,18 @@ def _load_from_checkpoint(
 
     if issubclass(cls, pl.LightningDataModule):
         return _load_state(cls, checkpoint, **kwargs)
-    return _load_state(cls, checkpoint, strict=strict, **kwargs)
+    # allow cls to be evaluated as subclassed LightningModule or,
+    # as LightningModule for internal tests
+    if issubclass(cls, pl.LightningModule) or isinstance(cls, pl.LightningModule):
+        return _load_state(cls, checkpoint, strict=strict, **kwargs)
```

Comment:
```suggestion
    return _load_state(cls, checkpoint, strict=strict, **kwargs)
```
I don't think this is required

Reply:
I can check again but this was added to troubleshoot failures in  test-pytorch-full previous to https://github.com/Lightning-AI/lightning/pull/13932/commits/ea0d812a3a4246316320447368a4d7e2dd159b3a

Reply:
sometimes test fails due to flaikness.. if a lot of them fail, then it could be an issue else we can re-run that specific test.

Reply:
the suggestion caused 

```sh
src/pytorch_lightning/core/saving.py:204: error: Argument 1 to "_load_state" has incompatible type "Type[ModelIO]"; expected "Union[Type[LightningModule], Type[LightningDataModule]]"  [arg-type]
```

...reverting